### PR TITLE
Fine Grained access rights

### LINKED ISSUE #1344: Cannot delete studies in production
Selecting study, then delete, then confirming does not delete the study 

expected behavior: studies are deleted after clicking delete and then confirm

### LINKED ISSUE #1437: Shall disable delete in template projects
## Long story short

Shall disable delete in template projects

image

Tried to delete this project (see above) but nothing happened ... then I noticed it was a template
I am not sure if the way to solve it is disabling delete action or simply separating more the templates from my own projects

### LINKED ISSUE #1459: change email address for "create account" popup on login page
Currently, clicking on "Create Account" on the osparc login page tells the user to contact "info@itis.swiss" which goes to a harassed Jacqueline. Can we change that to say "support@osparc.io"?

image

### LINKED ISSUE #1486: pressing on Export while on the main group shows no reaction
clickin on Export shows no feedback whatsoever.

I would expect either:
1. it works
2. Export is disabled.
3. a message appears.

### PR DESCRIPTION
**Pull request by odeimaiz**
#### Frontend
- Sharing resources with: (by @odeimaiz)
  - Everyone
  - My organization
  - Private
- fixes call to create as_template not passing the uuid (by @sanderegg)

#### Database: (by @sanderegg)
- modifications in projects table, prj_owner is still an email but now there is a a user id - foreign key instead of e-mail, add access_rights column containing JSON about gid:rights + creation/last modified date automatically set by postgres
- new groups table, contain groups id, name, description, type (standard, primary, everyone), creation/modification dates
- new user_to_groups table, contain links between user and groups, creation/modification dates
- modifications in users table, add primary_gid column
- added triggers in postgres that automatically create the primary group when a user is created
- added triggers in postgres that protect the all group from deletion
- migration scripts for all these points (migrate project table, migrate users table) through alembic

#### Webserver: (by @sanderegg)
- modification of the /me endpoint to retrieve the groups
- fixes cloning of project not changing the path to simcore-s3 files

### Bonus
- Templates and Studies are collapsible
- More uniform fonts throughout Study Browser
- added import-db recipe to allow copying a database from a remote host in order to test migration

## Related issue number
ITISFoundation/osparc-issues#22
closes #1437
closes #1344 
closes #1459
closes #1486

SaveAsTemplate

## Checklist

- [x] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [x] Unit tests for the changes exist
- [x] Runs in the swarm
- [x] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: services/web/client/source/class/osparc/auth/ui/LoginView.js [hunk 1]
```diff
@@ -100,7 +100,7 @@ qx.Class.define("osparc.auth.ui.LoginView", {
             if (config["invitation_required"]) {
               let text = this.tr("Registration is currently only available with an invitation.");
               text += "<br>";
-              text += this.tr("Please contact info@itis.swiss");
+              text += this.tr("Please contact support@osparc.io");
```
Comment (ignapas) [inline services/web/client/source/class/osparc/auth/ui/LoginView.js:103]: cherry-picked this into a ready to merge PR of mine

File: packages/postgres-database/docker/Dockerfile [hunk 2]
```diff
@@ -1,22 +1,51 @@
...
+WORKDIR /home/scu
+
+# bring installed package without build tools
+COPY --from=build ${VIRTUAL_ENV} ${VIRTUAL_ENV}
 
 # runtime dependency
-RUN apk add --no-cache \
-    bash \
-    postgresql
+# RUN apk add --no-cache \
```
Comment (pcrespov) [inline packages/postgres-database/docker/Dockerfile:46]: MINOR: rm or comment #

File: packages/postgres-database/src/simcore_postgres_database/migration/versions/53e095260441_add_projects_access_rights_and_.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    # FIXME: this does not work, alembic has no onupdate
+    op.alter_column(
+        "projects",
+        "last_change_date",
+        server_default=sa.text("now()"),
+        # onupdate=sa.text("now()"),
+    )
+    # define conversion fct
+    set_prj_owner_conversion_fct = sa.DDL(
+        f"""
```
Comment (pcrespov) [inline packages/postgres-database/src/simcore_postgres_database/migration/versions/53e095260441_add_projects_access_rights_and_.py:42]: TIP: can indent and use https://docs.python.org/3.6/library/textwrap.html to remove prefix spaces

File: packages/postgres-database/src/simcore_postgres_database/models/groups.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+import sqlalchemy as sa
+from sqlalchemy.sql import func
+
+from .base import metadata
+
+
+class GroupType(enum.Enum):
+    """
+        standard: standard group
```
Comment (pcrespov) [inline packages/postgres-database/src/simcore_postgres_database/models/groups.py:17]: can you define a bit more what is the different between standard and primary here?

File: packages/postgres-database/src/simcore_postgres_database/models/groups.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+
+# NOTE: using func.now() insted of python datetime ensure the time is computed server side
+groups = sa.Table(
+    "groups",
+    metadata,
+    sa.Column("gid", sa.BigInteger, nullable=False, primary_key=True),
+    sa.Column("name", sa.String, nullable=False),
+    sa.Column("description", sa.String, nullable=False),
+    sa.Column("type", sa.Enum(GroupType), nullable=False, server_default="STANDARD"),
```
Comment (pcrespov) [inline packages/postgres-database/src/simcore_postgres_database/models/groups.py:34]: should not server_default=GroupType.STANDARD?

File: packages/postgres-database/docker/Dockerfile (hunk 2 above)
Comment (sanderegg) [reply packages/postgres-database/docker/Dockerfile:46]: done

File: packages/postgres-database/src/simcore_postgres_database/models/groups.py (hunk 4 above)
Comment (sanderegg) [reply packages/postgres-database/src/simcore_postgres_database/models/groups.py:17]: yes done, basically this follows linux mechanism.
a primary group is the group created with the user. the user is typically the only one within that group. the primary group cannot be deleted unless the user is deleted.

File: packages/postgres-database/src/simcore_postgres_database/models/groups.py (hunk 5 above)
Comment (sanderegg) [reply packages/postgres-database/src/simcore_postgres_database/models/groups.py:34]: good point.

Comment (sanderegg) [reply packages/postgres-database/src/simcore_postgres_database/models/groups.py:34]: both works.

**Pull request merged**